packages/utilities/src/lib/hasAtLeastOneKeyInObject.ts

```diff
@@ -0,0 +1,22 @@
+import { isNullish } from './isNullOrUndefined';
+import type { NonNullObject } from './types';
+
+/**
+ * Checks whether any of the {@link keys} are in the {@link obj}
+ * @param obj - The object to check
+ * @param keys The keys to find in the object
+ * @returns `true` if at least one of the {@link keys} is in the {@link obj}, `false` otherwise.
+ *
+ * @example
+ * ```typescript
+ * const obj = { a: 1, b: 2, c: 3 };
+ * console.log(hasAtLeastOneKeyInObject(obj, ['a'])); // true
+ * console.log(hasAtLeastOneKeyInObject(obj, ['d'])); // false
+ * ```
+ */
+export function hasAtLeastOneKeyInObject<T extends NonNullObject, K extends keyof T>(
+	obj: T,
+	keys: readonly PropertyKey[]
+): obj is T & { [key in K]: NonNullable<T[K]> } {
```

Comment:
```suggestion
export function hasAtLeastOneKeyInObject<T extends NonNullObject, K extends PropertyKey>(
	obj: T,
	keys: readonly K[]
): obj is T & { [key in K]: NonNullable<T[K]> } {
```

now the typeguard will work properly, using `K extends PropertyKey & keyof T` allows any key and also gives auto complete for known keys but it breaks the typeguard, maybe u can experiment with it to make autocomplete and typeguard both work while also allowing any keys